modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/bond/DiscountingFixedCouponBondTradePricer.java

```diff
@@ -0,0 +1,1076 @@
+    double notional = product.getNotional();
+    PointSensitivityBuilder pvSensi = presentValueProductSensitivity(trade, provider).multipliedBy(1d / df / notional);
+    RepoCurveZeroRateSensitivity dfSensi = discountFactors.zeroRatePointSensitivity(settlementDate)
+        .multipliedBy(-pv.getAmount() / df / df / notional);
+    return pvSensi.combinedWith(dfSensi);
+  }
+
+  /**
+   * Calculates the dirty price sensitivity of the fixed coupon bond trade with z-spread.
+   * <p>
+   * The dirty price sensitivity of the trade is the sensitivity of the present value to
+   * the underlying curves.
+   * <p>
+   * The z-spread is a parallel shift applied to continuously compounded rates or periodic
+   * compounded rates of the discounting curve.
+   * <p>
+   * The computation is based on {@code settlementDate}. Thus {@code tradeInfo} in the trade must not be empty.
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param zSpread  the z-spread
+   * @param periodic  if true, the spread is added to periodic compounded rates,
+   *  if false, the spread is added to continuously compounded rates
+   * @param periodsPerYear  the number of periods per year
+   * @return the dirty price curve sensitivity of the trade
+   */
+  public PointSensitivityBuilder dirtyPriceSensitivityWithZspread(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double zSpread,
+      boolean periodic,
+      int periodsPerYear) {
+
+    FixedCouponBond product = trade.getProduct();
+    LocalDate settlementDate = trade.getTradeInfo().getSettlementDate().get();
+    StandardId securityId = trade.getSecurityLink().getStandardId();
+    StandardId legalEntityId = product.getLegalEntityId();
+    RepoCurveDiscountFactors discountFactors =
+        provider.repoCurveDiscountFactors(securityId, legalEntityId, product.getCurrency());
+    double df = discountFactors.discountFactor(settlementDate);
+    CurrencyAmount pv = presentValueProductWithZSpread(trade, provider, zSpread, periodic, periodsPerYear);
+    double notional = product.getNotional();
+    PointSensitivityBuilder pvSensi = presentValueProductSensitivityWithZSpread(
+        trade, provider, zSpread, periodic, periodsPerYear).multipliedBy(1d / df / notional);
+    RepoCurveZeroRateSensitivity dfSensi = discountFactors.zeroRatePointSensitivity(settlementDate)
+        .multipliedBy(-pv.getAmount() / df / df / notional);
+    return pvSensi.combinedWith(dfSensi);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the accrued interest of the fixed coupon bond trade.
+   * <p>
+   * The computation is based on {@code settlementDate}. Thus {@code tradeInfo} in the trade must not be empty.
+   * 
+   * @param trade  the trade to price
+   * @return the accrued interest of the trade 
+   */
+  public double accruedInterest(FixedCouponBondTrade trade) {
```

Comment:
Should have a version of "accrued interest" based on product and arbitrary date.